docs/blog/2019-09-26-gatsby-perf/index.md

```diff
@@ -0,0 +1,116 @@
+
+![WebPagetest v1](./images/v1-webpagetest.png)
+
+Armed with the powerful combo of Lighthouse, Webpagetest, and a knowledge of some useful performance terms like TTI, we can now shift into a discussion of the changes in Gatsby v2 that _improve_ performance over Gatsby v1 in meaningful, real-world ways. We'll come back to both of these tools as it relates to the Gatsby v2 site to actually _prove_ that we've improved performance over our Gatsby v1 baseline!
+
+## Improving upon Gatsby v1
+
+Gatsby v1 was in many ways an experiment to prove out some meaningful ideas. Injecting content at _build time_ from remote data sources with GraphQL. Statically rendering React components to HTML to maximize performance and SEO. Static site generation _and_ app-like functionality with React hydration. A pluggable and extensible architecture to augment and enhance Gatsby's base feature set and give Gatsby superpowers. [Pull data from Wordpress at build time][gatsby-source-wordpress]? Sure. [Author your application in Typescript][gatsby-plugin-typescript]? Seems reasonable. All this while maintaining sane and optimized defaults to truly squeeze every ounce of performance out of your application.
+
+Gatsby v2 set out to build upon this solid foundation, while focusing on improvements in speed and developer experience.
+
+### Speed improvements
+
+It's not 🚀 science. Shipping less JavaScript to your end users makes your application faster to load, parse, and use. Think _hard_ whether that slick launch animation and heavy above the fold hero image are actually _meaningful_ to the quality experience your users want. Every additional byte of JavaScript has an associated parse and evaluation time that you're forwarding along to your end users.
+
+> As much as I love JavaScript, it’s always the most expensive part of your site.
+> -- Addy Osmani
+
+To that end, Gatsby v2 ships 31% less JavaScript in its client runtime. While we'd _love_ to brag about this, most of the credit goes to smart optimizations in libraries we rely upon: React and @reach/router. Much like performance optimizations, build upgrades, etc. are available for free in something like [create-react-app][create-react-app], these upgrades in Gatsby v2 are available _for free_ simply by updating to Gatsby v2 by following the [migration guide][migration-guide]. The power of opinionated, optimized toolsets that internalize smart defaults!
+
+|    Library    | Gatsby v1 | Gatsby v2 |
+| :-----------: | :-------: | :-------: |
+|     React     |  49.8kb   |  34.8kb   |
+| @reach/router |    8kb    |    6kb    |
```

Comment:
not sure how to convey that - but we replaced both `react-router` AND `react-router-dom` with just `@reach/router` - so there is more bandwidth saving here

Reply:
Oh nice! Yeah, I can make that more explicit.

Reply:
https://bundlephobia.com/result?p=react-router@4.3.1 & https://bundlephobia.com/result?p=@reach/router@1.2.0

Reply:
You suggesting just linking to those? I like that.